azure-nextgen-cs-synapse/MyStack.cs

```diff
@@ -0,0 +1,133 @@
+// Copyright 2016-2020, Pulumi Corporation.  All rights reserved.
+
+using Pulumi;
+using Pulumi.AzureNextGen.Authorization.V20200401Preview;
+using Pulumi.AzureNextGen.Resources.Latest;
+using Pulumi.AzureNextGen.Storage.Latest;
+using Pulumi.AzureNextGen.Synapse.V20190601Preview;
+using Pulumi.AzureNextGen.Synapse.V20190601Preview.Inputs;
+using Pulumi.Random;
+using SkuArgs = Pulumi.AzureNextGen.Storage.Latest.Inputs.SkuArgs;
+
+class MyStack : Stack
+{
+    public MyStack()
+    {
+        var config = new Pulumi.Config();
+        var location = config.Get("location") ?? "WestUS";
+
+        var resourceGroup = new ResourceGroup("resourceGroup", new ResourceGroupArgs
+        {
+            ResourceGroupName = "synapse-rg",
+            Location = location
+        });
+
+        var storageAccount = new StorageAccount("storageAccount", new StorageAccountArgs
+        {
+            ResourceGroupName = resourceGroup.Name,
+            Location = resourceGroup.Location,
+            AccountName = "synapsesa",
+            AccessTier = "Hot",
+            EnableHttpsTrafficOnly = true,
+            IsHnsEnabled = true,
+            Kind = "StorageV2",
+            Sku = new SkuArgs
+            {
+                Name = "Standard_RAGRS"
+            },
+        });
+        var dataLakeStorageAccountUrl = Output.Format($"https://{storageAccount.Name}.dfs.core.windows.net");
+        
+        var users = new BlobContainer("users", new BlobContainerArgs
+        {
+            ResourceGroupName = resourceGroup.Name,
+            AccountName = storageAccount.Name,
+            ContainerName = "users",
+            PublicAccess = "None"
+        });
+        
+        var workspace = new Workspace("workspace", new WorkspaceArgs
+        {
+            ResourceGroupName = resourceGroup.Name,
+            Location = resourceGroup.Location,
+            WorkspaceName = "my-workspace",
+            DefaultDataLakeStorage = new DataLakeStorageAccountDetailsArgs
+            {
+                AccountUrl = dataLakeStorageAccountUrl,
+                Filesystem = "users"
+            },
+            Identity = new ManagedIdentityArgs
+            {
+                Type = "SystemAssigned"
+            },
+            SqlAdministratorLogin = "sqladminuser",
+            SqlAdministratorLoginPassword = "YourStrongPassword"
+        });
+
+        var allowAll = new IpFirewallRule("allowAll", new IpFirewallRuleArgs
+        {
+            ResourceGroupName = resourceGroup.Name,
+            WorkspaceName = workspace.Name,
+            RuleName = "allowAll",
+            EndIpAddress = "255.255.255.255",
+            StartIpAddress = "0.0.0.0"
+        });
+
+        var subscriptionId = resourceGroup.Id.Apply(id => id.Split('/')[2]);
+        var roleDefinitionId = $"/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/roleDefinitions/ba92f5b4-2d11-453d-a403-e96b0029c9fe";
```

Comment:
Is this something we need to make secret?

Reply:
No, this is a "well-known" GUID of a role defined by Microsoft